monitoring/opencensus/metrics.go

```diff
@@ -5,106 +5,68 @@ import (
-// Datadog exporter assumes the Datadog Agent is running on localhost
-//TODO(dazwilkin) Is this the best place to initialize the exporters?
-//TODO(dazwilkin) OpenCensus exporters should be configured by config
-func init() {
-	// Stackdriver Exporter
-	sd, err := stackdriver.NewExporter(stackdriver.Options{
-		// MetricPrefix helps uniquely identify these metrics
-		//TODO(dazwilkin) How to create the exporter in order to use MetricFactory.Prefix
-		MetricPrefix: namespace,
-	})
-	if err != nil {
-		log.Fatal(err)
-	}
-	// Important to invoke Flush before exiting
-	defer sd.Flush()
-
-	// Prometheus Exporter
-	// Provides an http.Handler for the metrics endpoint
-	//TODO(dazwilkin) How to provide this handler back to the Trillian service?
-	// e.g. mux.Handle("/metrics", pm)
-	pm, err := prometheus.NewExporter(prometheus.Options{
-		Namespace: namespace,
-	})
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	// Datadog Exporter
-	// Assumes the Datadog agent is running
-	dd, err := datadog.NewExporter(datadog.Options{
-		Namespace: namespace,
-	})
-	if err != nil {
-		log.Fatal(err)
+// Initialize is called by Trillian Servers to configure OpenCensus Agent Exporter.
+// See https://github.com/google/trillian/pull/1414#pullrequestreview-195485927
+// Once registered with Opencensus' View, all metrics will export to these systems
+//TODO(dazwilkin) OpenCensus Agent Exporter should be configured by config
+//TODO(dazwilkin) Address 'WithInsecure()'
+func Initialize() (func(), error) {
+	agent, err := ocagent.NewExporter(
+		ocagent.WithInsecure(),
+		ocagent.WithServiceName(fmt.Sprintf("trillian-%d", os.Getpid())))
+	if err == nil {
+		trace.RegisterExporter(agent)
+		view.RegisterExporter(agent)
 	}
-	defer dd.Stop()
-
-	// Register Exporter(s) as metrics exporter(s)
-	view.RegisterExporter(sd)
-	view.RegisterExporter(pm)
-	view.RegisterExporter(dd)
-
-	// Stackdriver requires 60s reporting period
-	view.SetReportingPeriod(60 * time.Second)
+	return func() {
+		agent.Stop()
+	}, err
 }
 
-// checkLabelNames as required by OpenCensus fails if any label name
+// checkLabelNames as required by OpenCensus fails if any label name:
 // -- contains non-printable ASCII
 // -- or len is 0 or >256
 // Printable ASCII 32-126 inclusive
-func checkLabelNames(labelNames []string) {
+func checkLabelNames(names []string) {
 	nonPrintableASCII := func(r rune) bool { return (r < 32 || r > 126) }
-	for _, labelName := range labelNames {
-		if len(labelName) == 0 || len(labelName) > 256 {
+	for _, name := range names {
+		if len(name) == 0 || len(name) > 256 {
 			glog.Fatalf("OpenCensus label names must be between 1 and 256 characters")
 		}
-		if strings.IndexFunc(labelName, nonPrintableASCII) != -1 {
-			glog.Fatalf("OpenCensus label names must be printable ASCII; '%s' is not", labelName)
+		if strings.IndexFunc(name, nonPrintableASCII) != -1 {
+			glog.Fatalf("OpenCensus label names must be printable ASCII; '%s' is not", name)
```

Comment:
nitty nit: can use `%q` rather than `'%s'` for the same output.

Reply:
Thanks!

I did not know that... my printf foo is weak.

Fixed: https://github.com/google/trillian/pull/1414/commits/8986d5a80a9e7b9c0053b3fd384319cf9bf6832f